sdk/core/core-http/src/fetchHttpClient.ts

```diff
@@ -66,6 +69,28 @@ export abstract class FetchHttpClient implements HttpClient {
         }
       };
       httpRequest.abortSignal.addEventListener("abort", abortListener);
+
+      registerCleanUpListenerOnStream = (stream: NodeJS.ReadableStream, isDownload: boolean) => {
+        const cleanupCallback = () => {
+          if (isDownload) {
+            streamingDownload = false;
```

Comment:
I don't like how this callback manipulates variables in the outer scope, especially because they are declared below this function definition.